### PR TITLE
Add Mono.Options.CommandSet

### PR DESCRIPTION
**Pull request by jonpryor**
`Mono.Options.CommandSet` allows easily having separate commands and
associated command options, allowing creation of a *suite* along the
lines of **git**(1), **svn**(1), etc.

`CommandSet` allows intermixing plain text strings for `--help` output,
Option values -- as supported by `OptionSet` -- and `Command` instances,
which have a name, optional help text, and an optional `OptionSet`.

	var suite = new CommandSet ("suite-name") {
		// Use strings and option values, as with OptionSet
		"usage: suite-name COMMAND [OPTIONS]+",
		{ "v:", "verbosity", (int? v) => Verbosity = v.HasValue ? v.Value : Verbosity+1 },
		// Commands may also be specified
		new Command ("command-name", "command help") {
			Options = new OptionSet {/*...*/},
			Run     = args => { /*...*/},
		},
		new MyCommandSubclass (),
	};
	suite.Run (new string[]{...});

`CommandSet` provides a `help` command, and forwards `help COMMAND`
to the registered `Command` instance by invoking `Command.Invoke()`
with `--help` as an option.

### CODE REVIEW

File: mcs/class/Mono.Options/Mono.Options-net_4_x.csproj [hunk 1]
```diff
@@ -52,7 +52,24 @@
     <Compile Include="..\..\build\common\Consts.cs" />
```
Comment (marek-safar) [inline mcs/class/Mono.Options/Mono.Options-net_4_x.csproj]: This file is auto generated

File: mcs/class/Mono.Options/Makefile [hunk 2]
```diff
@@ -17,6 +17,16 @@ mono_options_DATA = Mono.Options/Options.cs
 
 include ../../build/library.make
 
+test-local: Mono.Options-PCL.dll
+
+clean-local: clean-pcl
+
+Mono.Options-PCL.dll: Mono.Options.dll.sources $(shell cat Mono.Options.dll.sources)
+	$(CSCOMPILE) -target:library -out:$@ -debug+ -d:PCL -lib:../lib/$(PROFILE) -r:System.dll @$<
```
Comment (marek-safar) [inline mcs/class/Mono.Options/Makefile]: Does this compile with master?

File: mcs/class/Mono.Options/Makefile [hunk 3]
```diff
@@ -47,7 +58,7 @@ Documentation/en/examples/Mono.Options.dll: $(the_lib)
 	-cp $^.mdb $@.mdb
 
 %.exe: %.cs Documentation/en/examples/Mono.Options.dll
-	$(CSCOMPILE) -debug+ -r:Mono.Posix.dll -r:System.Core.dll -lib:Documentation/en/examples -r:Mono.Options.dll -out:$@ $<
+	$(CSCOMPILE) -debug+ -lib:../lib/$(PROFILE) -r:System.dll -r:Mono.Posix.dll -r:System.Core.dll -lib:Documentation/en/examples -r:Mono.Options.dll -out:$@ $<
```
Comment (marek-safar) [inline mcs/class/Mono.Options/Makefile]: Don't use lib but full path with `-r`

File: mcs/class/Mono.Options/Test/packages.config [hunk 4]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (marek-safar) [inline mcs/class/Mono.Options/Test/packages.config]: I think for mono code we always use nunitlite, not sure we need to be different here

File: mcs/class/Mono.Options/Makefile (hunk 2 above)
Comment (jonpryor) [inline mcs/class/Mono.Options/Makefile]: PR is based on mono/2a239b52, which is master circa February 27...

**Pull request merged**